### DISCUSSION TITLE
Error During IPA Build for Distribution

### DISCUSSION THREAD

**Discussion by Rustam-Hashymov** [codemagic-ci-cd/codemagic-docs#2820] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1. Run the build pipeline on Codemagic to create the IPA for distribution.
2. Codemagic initiates the build process and executes the scripts as configured.

### Expected results

The IPA build should complete successfully without any errors, creating an archive ready for distribution, and the app should be published as expected.

### Actual results

The build fails with the following error message:
`Failed to archive /Users/builder/clone/ios/Eridan_Duplicator_Mobile_App.xcworkspace. The following build commands failed:

CompileC /Users/builder/Library/Developer/Xcode/DerivedData/Eridan_Duplicator_Mobile_App-hcbnencqhkgxhtaiqfpwqzfzkqeo/Build/Intermediates.noindex/ArchiveIntermediates/stage/IntermediateBuildFilesPath/Eridan_Duplicator_Mobile_App.build/Release-iphoneos/Eridan_Duplicator_Mobile_App_Stage.build/Objects-normal/arm64/AppDelegate.o /Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/AppDelegate.mm normal arm64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
	...
	/Applications/Xcode-16.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -ivfsstatcache /Users/builder/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphoneos18.0-22A3362-77c87d3582361da8819bf4750a9bd04a.sdkstatcache -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -fno-color-diagnostics -fmodules-prune-interval\=86400 -fmodules-prune-after\=345600 -fbuild-session-file\=/Users/builder/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror\=non-modular-include-in-framework-module -Wno-trigraphs -Wno-missing-field-initializers -Wno-missing-prototypes -Werror\=return-type -Wunreachable-code -Wquoted-include-in-framework-header -Wno-implicit-atomic-properties -Werror\=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror\=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-c++11-extensions -Wno-implicit-fallthrough -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body @/Users/builder/Library/Developer/Xcode/DerivedData/Eridan_Duplicator_Mobile_App-hcbnencqhkgxhtaiqfpwqzfzkqeo/Build/Intermediates.noindex/ArchiveIntermediates/stage/IntermediateBuildFilesPath/Eridan_Duplicator_Mobile_App.build/Release-iphoneos/Eridan_Duplicator_Mobile_App_Stage.build/Objects-normal/arm64/af3fcb34312c57c0f52879cdce924b91-common-args.resp -MMD -MT dependencies -MF /Users/builder/Library/Developer/Xcode/DerivedData/Eridan_Duplicator_Mobile_App-hcbnencqhkgxhtaiqfpwqzfzkqeo/Build/Intermediates.noindex/ArchiveIntermediates/stage/IntermediateBuildFilesPath/Eridan_Duplicator_Mobile_App.build/Release-iphoneos/Eridan_Duplicator_Mobile_App_Stage.build/Objects-normal/arm64/AppDelegate.d --serialize-diagnostics /Users/builder/Library/Developer/Xcode/DerivedData/Eridan_Duplicator_Mobile_App-hcbnencqhkgxhtaiqfpwqzfzkqeo/Build/Intermediates.noindex/ArchiveIntermediates/stage/IntermediateBuildFilesPath/Eridan_Duplicator_Mobile_App.build/Release-iphoneos/Eridan_Duplicator_Mobile_App_Stage.build/Objects-normal/arm64/AppDelegate.dia -c /Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/AppDelegate.mm -o /Users/builder/Library/Developer/Xcode/DerivedData/Eridan_Duplicator_Mobile_App-hcbnencqhkgxhtaiqfpwqzfzkqeo/Build/Intermediates.noindex/ArchiveIntermediates/stage/IntermediateBuildFilesPath/Eridan_Duplicator_Mobile_App.build/Release-iphoneos/Eridan_Duplicator_Mobile_App_Stage.build/Objects-normal/arm64/AppDelegate.o
	In file included from /Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/AppDelegate.mm:1:
	/Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/AppDelegate.h:1:9: fatal error: 'RCTAppDelegate.h' file not found
	1 | #import <RCTAppDelegate.h>
	|         ^~~~~~~~~~~~~~~~~~
	1 error generated.

ScanDependencies /Users/builder/Library/Developer/Xcode/DerivedData/Eridan_Duplicator_Mobile_App-hcbnencqhkgxhtaiqfpwqzfzkqeo/Build/Intermediates.noindex/ArchiveIntermediates/stage/IntermediateBuildFilesPath/Eridan_Duplicator_Mobile_App.build/Release-iphoneos/Eridan_Duplicator_Mobile_App_Stage.build/Objects-normal/arm64/main.o /Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/main.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
	...
	/Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/AppDelegate.h:1:9: error: 'RCTAppDelegate.h' file not found (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
	/Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/main.m:3:9: note: in file included from /Users/builder/clone/ios/Eridan_Duplicator_Mobile_App/main.m:3:
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
	warning: Run script build phase 'Start Packager' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
	warning: Run script build phase 'Bundle React Native code and images' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
	warning: Run script build phase 'Upload Debug Symbols to Sentry' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Eridan_Duplicator_Mobile_App_Stage' from project 'Eridan_Duplicator_Mobile_App')
`

After this error, other scripts continue to run and report a "success" status. However, the app does not get published as intended, likely due to the initial build failure.

**Comment by dam-ease** [comment]
Hello @Rustam-Hashymov.
From the error message, the issue seems related to react native projects, can you check out the suggested solutions in this SO thread - https://stackoverflow.com/questions/75822104/rctappdelegate-h-file-not-found
Also, Can you please share the build ID with me? You can find it under your build details page's 'ID' section.
﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Reply by Rustam-Hashymov** [reply]
Hi, its my build id - 67331abec7799733fa8f1788

**Reply by dam-ease** [reply]
Thanks for sharing!
Have you tried the suggested solutions in the thread above? This issue doesn't seem specific to building with codemagic, are you able to build your ipa successfully locally?
Kindly try the solutions in below threads as well:
- https://stackoverflow.com/questions/75822104/rctappdelegate-h-file-not-found
- https://github.com/facebook/react-native/issues/42654
Please let me know if you run into further issues.

**Reply by dam-ease** [reply]
Hello @Rustam-Hashymov,
Any update for me on this? If there is anything else I can assist you with, please don't hesitate to let me know.

**Reply by dam-ease** [reply]
I see your recent builds have been successful.
I'll be closing this thread for now.
Happy building!
